compaction/time_window_compaction_strategy.cc

```diff
@@ -431,12 +431,10 @@ time_window_compaction_strategy::get_compaction_candidates(table_state& table_s,
 timestamp_type
 time_window_compaction_strategy::get_window_lower_bound(std::chrono::seconds sstable_window_size, timestamp_type timestamp) {
     using namespace std::chrono;
-    auto timestamp_in_sec = duration_cast<seconds>(microseconds(timestamp)).count();
-
     // mask out window size from timestamp to get lower bound of its window
-    auto window_lower_bound_in_sec = seconds(timestamp_in_sec - (timestamp_in_sec % sstable_window_size.count()));
+    auto num_windows = microseconds(timestamp) / sstable_window_size;
 
-    return timestamp_type(duration_cast<microseconds>(window_lower_bound_in_sec).count());
+    return duration_cast<microseconds>(num_windows * sstable_window_size).count();
```

Comment:
So we exchange `x - (x % y)` with `(x / y) * y`, right?
The sub+mod are cheaper instructions as compared to div+mul, do we trust the compiler smartness?

Reply:
> So we exchange `x - (x % y)` with `(x / y) * y`, right? 

right

> The sub+mod are cheaper instructions as compared to div+mul, do we trust the compiler smartness?

the reason i prefer div + mul is that it looks simpler. but yes, i agree with you that sub + mod could be more performant. but i don't consider this function a bottle neck. so i favor readability over performance here. but if you think otherwise, i will switch back to the original math.

please let me know.

Reply:
I don't have strong preference. @raphaelsc , WDYT?

Reply:
To be honest, I have to think more to understand the new formula since it plays with two different time units when calculating num of windows. I may be biased though.

Reply:
@raphaelsc ping?

Reply:
@raphaelsc ping?

Reply:
@tchaikov I assume the division of different unit promotes the one with coarser granularity to the more fine-grained one right?

so
microseconds(timestamp) / sstable_window_size

is equivalent to

microseconds(timestamp) / microseconds(sstable_window_size)

?

Reply:
I think we can apply the patch. /cc @denesb

Reply:
> @tchaikov I assume the division of different unit promotes the one with coarser granularity to the more fine-grained one right?
> 
> so microseconds(timestamp) / sstable_window_size
> 
> is equivalent to
> 
> microseconds(timestamp) / microseconds(sstable_window_size)
> 
> ?

yes, see https://godbolt.org/z/5TvaYa3n6

Reply:
thanks!